wiring/inc/spark_wiring_ble.h

```diff
@@ -253,7 +253,12 @@ class BleUuid {
     void construct(const char* uuid);
     void toBigEndian(uint8_t buf[BLE_SIG_UUID_128BIT_LEN]) const;
 
-    hal_ble_uuid_t uuid_;
+    const uint8_t BASE_UUID[BLE_SIG_UUID_128BIT_LEN] = {0xFB,0x34,0x9B,0x5F,0x80,0x00, 0x00,0x80, 0x00,0x10, 0x00,0x00, 0x00,0x00,0x00,0x00};
+    const uint8_t UUID16_LO = 12;
+    const uint8_t UUID16_HI = 13;
+
+    uint8_t uuid128_[BLE_SIG_UUID_128BIT_LEN];
```

Comment:
Any reason not to keep using `hal_ble_uuid_t` and using `BLE_UUID_TYPE_128BIT_SHORTED` to indicate that it's a 16-bit UUID expanded to 128 bit to avoid unnecessary copying?

Reply:
Since the 16-bits UUID and 128-bits UUID is organized in a union struct. And in BLE HAL where a `hal_ble_uuid_t` being referenced it just uses `.uuid16` to get shorted UUID, which will actually return the lowest two bytes of the expanded 128-bits UUID. If we just pass the private `hal_ble_uuid_t` member to BLE HAL, it would result incorrect 16-bits UUID in BLE HAL. And FYI, the `BLE_UUID_TYPE_128BIT_SHORTED` is actually a temporary hack to resolve unknown 128-bits vendor UUID, so that BLE HAL can retrieve complete 128-bits UUID afterwards before exposing it to user.

Reply:
We can also use `hal_ble_uuid_t` as a member of wiring BLE UUID class. But we need to refactor `hal_ble_uuid_t` to be that 16-bits UUID and 128-bits UUID are organized as separated member of `hal_ble_uuid_t`. Does it make sense?

Reply:
I think it's fine to leave things as-is, but I'm concerned with too much copying going on.

Is there any problem with using `BLE_UUID_TYPE_128BIT_SHORTED` as a convention in both wiring and HAL that it's a 16-bit UUID expanded to 128 bits? For simplicity in wiring we'd always be expanding 16-bit UUIDs into this type.